### ISSUE TITLE
SA RepairGTA is Deleting the Unlocked Girlfriend Cars with Custom Plates from Players' Saves.

### ISSUE THREAD

**Issue by OrionSR** [JuniorDjjr/RepairGTA#1]
`static void ShowCarGeneratorConsiderationMessageIfNeeded(CCarGenerator carGenerator)`

This section suggests you were aware of an issue but not the implications. The unlocked girlfriend cars should not be removed; they are needed when girlfriend progress is above 50%. 

Also, there are two unused Greenwoods and a BMX cargen at the Jefferson Hotel that are never displayed. This tool is removing the second Greenwood. Although these vehicles aren't needed, they are well known and occasionally repurposed by players. Having a Greenwood missing leaves a gap in the standard block of SCM cargens that reduces the effectiveness of other repair tools, and throws errors when the expected number of cargens is not found in the save.

Please update this tool so no standard SCM cargens are removed.

**Comment by JuniorDjjr** [comment]
> The unlocked girlfriend cars should not be removed

How can you see this problem happening? This only happens at the 50% transition (two cars are used, before and after the 50%?) because here I don't notice the bug (it's unlocked).


> that reduces the effectiveness of other repair tools

What repair tools for example?

**Comment by JuniorDjjr** [comment]
Released: https://www.mixmods.com.br/2023/08/atualizados-cleo-cleo-repairgta/

https://github.com/JuniorDjjr/RepairGTA/commit/537815ed0ce323400a5165386bf0c813859c31e4

**Comment by OrionSR** [comment]
Oh wow. I wasn't expecting such a quick response. I know you've been quite busy recently. Thank you.

Yeah, this issue is hard to notice unless players progress the girlfriends above the 50% mark. I've named this glitch the 201 Cargen Glitch, because the 010 template I use to analyze save files gives an alert count whenever the number of cargens is anything other that 208.

Prediction: If the unlocked (at 50% progress) girlfriend cargens are removed, it's possible the player may notice their custom plates being applied to streamed cargens parked alongside the roads or in small parking lots, or to their added custom cargens.

CarGenFix is a rather crude (no safeties for size of main, etc) cleo script that removes all cargens before and after the standard block of SCM cargens. This should remove all non-standard cargens, unless there's a gap in the middle of the SCM cargen block.
[CarGenFix.zip](https://github.com/JuniorDjjr/RepairGTA/files/12337700/CarGenFix.zip)

Please keep this issue open for a while. I'm working on a script to replace the missing cargens to their original slots, and would like to post the script here for anyone who manages to track the glitch to it's source.

**Comment by OrionSR** [comment]
**201Fix.cs** - a cleo script that replaces the standard car generators removed as duplicates by older versions of RepairGTA.
Download: [201Fix.zip](https://github.com/JuniorDjjr/RepairGTA/files/12387422/201Fix.zip)

If your girlfriends' cars are missing after reaching 50% progress then this script will replace them.

Type "201Fix" to begin the repair.
Type "201Info" to redisplay the car generator report that pops up when the script first starts.

For more details on the glitch, and about how to install and use this script, see Freeman39's [excellent documentation](https://gtaforums.com/topic/992132-girlfriend-cars-missing-after-completing-the-game/?do=findComment&comment=1072199510).

The forum users asked me to pass on their thanks for the quick update to this popular tool.

BTW, the description about the update has the glitch backwards. The cars that were removed are displayed only after reaching 50% progress with the girlfriend. Before 50% progress, the cars that are displayed have locked doors. After 50% the doors are unlocked, and on standard games, they have custom plates (except the Bandito has no plates at all).

**Comment by JuniorDjjr** [comment]
It's also possible to implement the fix inside RepairGTA itself (first checking if it's really using vanilla storyline to avoid creating in total conversions mod etc), but I think isn't required, I just added the script on RepairGTA post.
